src/Parser/DemonHunter/Havoc/Modules/Items/AngerOfTheHalfGiants.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+import Wrapper from 'common/Wrapper';
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+import FuryTracker from '../ResourceTracker/FuryTracker';
+
+/**
+* Anger Of The Half Giants
+* Equip: Demon Blades/Demon Blades
+* Demon's Bite/Demon Blades generates an additional 1 to 14 Fury.
+**/
+
+
+class AngerOfTheHalfGiants extends Analyzer {
+	static dependencies = {
+		furyTracker: FuryTracker,
+		abilityTracker: AbilityTracker,
+		combatants: Combatants,
+	};
+
+	_hasDemonBlades = false;
+
+	on_initialized() {
+		this.active = this.combatants.selected.hasFinger(ITEMS.ANGER_OF_THE_HALF_GIANTS.id);
+		if(this.active) {
+			this._hasDemonBlades = this.combatants.selected.hasTalent(SPELLS.DEMON_BLADES_TALENT.id);
+		}
+	}
+
+	get angerFury() {
+		return this.furyTracker.buildersObj[SPELLS.ANGER_OF_THE_HALF_GIANTS_FURY.id];
+	}
+
+	get dBCasts() {
+		if(!this._hasDemonBlades) {
+			return this.abilityTracker.getAbility(SPELLS.DEMONS_BITE.id).casts;
+		}
+		if(this.furyTracker.buildersObj[SPELLS.DEMON_BLADES_FURY.id]) {
+			return this.furyTracker.buildersObj[SPELLS.DEMON_BLADES_FURY.id].casts;
+		}
+		return 0;
+	}
+
+	get furyGenerated() {
+		if(this.angerFury){
+			return this.angerFury.generated;
+		}
+		return 0;
+	}
+
+	get furyWasted() {
+		if(this.angerFury){
+			return this.angerFury.wasted;
+		}
+		return 0;
+	}
+
+	get totalFury() {
+		return this.furyGenerated + this.furyWasted;
+	}
+
+	item() {
+		const builderId = this._hasDemonBlades ? SPELLS.DEMON_BLADES_TALENT.id : SPELLS.DEMONS_BITE.id;
+		return {
+			item: ITEMS.ANGER_OF_THE_HALF_GIANTS,
+			result: (
+				<dfn data-tip={`Total Fury Gained: <b>${formatNumber(this.totalFury)}</b>.`}>
```

Comment:
Should show actual gain, not the amount would have gotten with perfect play (so show `this.furyGenerated` instead of `this.totalFury`)